src/Runner/FileCachingLintingFileIterator.php

```diff
@@ -24,19 +24,11 @@
  *
  * @extends \CachingIterator<mixed, \SplFileInfo, \Iterator<mixed, \SplFileInfo>>
  */
-final class FileCachingLintingIterator extends \CachingIterator
+final class FileCachingLintingFileIterator extends \CachingIterator implements LintingResultAwareFileIteratorInterface
 {
     private LinterInterface $linter;
-
-    /**
-     * @var LintingResultInterface
-     */
-    private $currentResult;
-
-    /**
-     * @var LintingResultInterface
-     */
-    private $nextResult;
+    private ?LintingResultInterface $currentResult = null;
```

Comment:
I am not so sure if it would be so fast 😅. This is a product of some refactoring and I just did not want to extract it from here because it could complicate branch's maintenance.